src/Tooltip/Tooltip.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+
+import * as React from 'react';
+import { Dimensions, View } from 'react-native';
+import type { ViewLayoutEvent } from 'react-native/Libraries/Components/View/ViewPropTypes';
+import TooltipBubble from './TooltipBubble';
+import StyleSheet from '../PlatformStyleSheet';
+
+type Props = {|
+  +isActive?: boolean,
+  +arrowHorizontalPosition?: 'flex-start' | 'center' | 'flex-end',
+  +children: React.Node,
+  +text: string,
+  +verticalShift?: number,
+|};
+
+type State = {|
+  elementY: number,
+  childrenHeight: number,
+|};
+
+export default class Tooltip extends React.Component<Props, State> {
+  childrenRef: ?React$ElementRef<Object>;
+
+  static defaultProps = {
+    isActive: false,
+    arrowHorizontalPosition: 'center',
+    verticalShift: 0,
+  };
+
+  state = {
+    elementY: 0,
+    childrenHeight: 0,
+  };
+
+  componentDidUpdate() {
+    this.measureChildrenHeight();
+  }
+
+  measureChildrenHeight = () => {
+    this.childrenRef &&
+      this.childrenRef.measure((a, b, width, height) => {
+        const roundedHeight = Math.round(height);
+        const { childrenHeight } = this.state;
+        roundedHeight !== childrenHeight &&
+          this.setState({ childrenHeight: roundedHeight });
+      });
+  };
+
+  onLayout = (event: ViewLayoutEvent) => {
+    const { layout } = event.nativeEvent;
```

Comment:
I don't know if this is necessary, but it could be worth it to setState only if the value is different. What do you think?